### PR TITLE
feat: Validate user API keys in CLI

### PR DESCRIPTION
**Pull request by MusabShakeel576**
This solution involves:
- [x]  Implement a key validation function within the setup.sh/newEnvQuestions.js.
- [x] Test the new function to ensure its effectiveness and reliability. (There is no testing tool setup in the `cli` folder, I did manual testing, should I set up one or does manual testing suffice for now?)
- [x] Update the setup documentation to inform users of this new feature. (Already mentioned in [Interactive Setup](https://docs.reworkd.ai/development/setup#interactive-setup))

Issue: #892

### CODE REVIEW

**Comment by asim-shrestha** [comment]
Thanks @MusabShakeel576! 

As per testing, I was thinking we could even make an example call to their endpoints here to see if if works. This should suffice for now though as long as the google serp key is flagged by this!

**Comment by asim-shrestha** [comment]
Had to update the replicate validator to the following:

`return isValidKey(apikey, /^r8_[a-zA-Z0-9]{37}$/)`

So `r8_` instead of `r8-`. Do you have a key with `r8-`?

File: cli/src/helpers.js [hunk 1]
```diff
@@ -18,8 +18,11 @@ export const printTitle = () => {
   );
 };
 
-// Function to check if entered api key is in the correct format
-export const isValidSkKey = (apikey) => {
-  const pattern = /^sk-[a-zA-Z0-9]{48}$/;
-  return pattern.test(apikey);
+// Function to check if entered api key is in the correct format or empty
+export const isValidKey = (apikey, pattern) => {
+  if (pattern.test(apikey) || apikey === "") {
```
Comment (awtkns) [inline cli/src/helpers.js:23]:
```suggestion
  if (apikey === "" || pattern.test(apikey)) {
```

**Review by awtkns — APPROVED** [review]
looks good to me once Asim's changes are addressed

**Comment by MusabShakeel576** [comment]
@asim-shrestha added example API calls, and update Replicate key.

File: cli/src/helpers.js (hunk 1 above)
Comment (MusabShakeel576) [reply cli/src/helpers.js]: @awtkns that's a nice catch to prevent extra computation, I fixed it.

**Comment by asim-shrestha** [comment]
Awesome @MusabShakeel576! 

Made the following change to fix some issues:
`🐛 Add node-fetch and allow api keys to be empty strings`

Thanks for the help here! This will be really valuable for new devs

**Pull request merged**